examples/qtsupport/qt_group_chat.cpp

```diff
@@ -26,57 +26,62 @@ CAF_PUSH_WARNINGS
 #include "ui_chatwindow.h" // auto generated from chatwindow.ui
 CAF_POP_WARNINGS
 
+#include "chatwidget.hpp"
+
 using namespace std;
 using namespace caf;
 
 class config : public actor_system_config {
 public:
-  std::string name;
-  std::string group_id;
-
-  config(int argc, char** argv) {
+  config() {
     opt_group{custom_options_, "global"}
-    .add(name, "name,n", "set name")
-    .add(group_id, "group,g", "join group (format: <module>:<id>");
-    parse(argc, argv);
-    load<io::middleman>();
+      .add<std::string>("name,n", "set name")
+      .add<std::string>("group,g", "join group");
   }
 };
 
 int main(int argc, char** argv) {
-  config cfg{argc, argv};
-  actor_system system{cfg};
-  auto name = cfg.name;
+  init_global_meta_objects<id_block::qt_support>();
+  core::init_global_meta_objects();
+  io::middleman::init_global_meta_objects();
+  config cfg{};
+  auto err = cfg.parse(argc, argv);
+  if (cfg.cli_helptext_printed)
+    return 0;
+  cfg.load<io::middleman>();
+  actor_system sys{cfg};
```

Comment:
Please use the `CAF_MAIN` scaffold here instead of making all of these steps by hand. This example simple hasn't been ported to it yet.

Reply:
I will port it to `CAF_MAIN`.